### PR TITLE
[docs-only] Create README for settings service

### PR DESCRIPTION
**Pull request by rhafer**
Fixes: #6295 (The settings service dev docs content needs a rewrite (admin docs relevant))

The just moves over the contents from  docs/services/settings/_index.md.

### CODE REVIEW

**Comment by rhafer** [comment]
@kobergj Am I right to assume that the dev docs index page for settings (https://owncloud.dev/services/settings/) will be regenerated from this PR once it is merged?

File: services/settings/README.md [hunk 1]
```diff
@@ -31,7 +21,3 @@ The diagram shows how the settings service integrates into oCIS:
 
 **Settings usage:**
 - Extensions can query ocis-settings for *settings values* of a user.
```
Comment (mmattel) [inline services/settings/README.md:23]:
```suggestion
## Settings Usage
Services can query ocis-settings for *settings values* of a user.
```

**Comment by mmattel** [comment]
> assume that the dev docs index page for settings will be regenerated from this PR

Just locally tested and as expected it gets regenerated 😃

**Comment by kobergj** [comment]
@rhafer yes. Doc pages will be created automatically from the `README.md`

**Comment by mmattel** [comment]
There are issues with the mermaid image:

* Though you can fix the codacy complains by defining this block as mermaid, it will then no longer be rendered as such in hugo. This needs a different approach.
* The mermaid content itself is wrong as it still uses the accounts service which was removed last year.

The cache definition text is imho wrong. On top you write about `SETTINGS_CACHE_STORE` but below in item 4 you define `STORAGE_USERS_STAT_CACHE_STORE_NODES` and
`STORAGE_USERS_FILEMETADATA_CACHE_STORE_NODES`. Beside the envvars used here (storage vs settings) if you have two items in redis-sentinel, you also must provide two on top. This is a 1:1 match.

**Comment by rhafer** [comment]
> * Though you can fix the codacy complains by defining this block as mermaid, it will then no longer be rendered as such in hugo. This needs a different approach
The mermaid diagram would still be rendered correctly. `hugo` know abouth the ` ```mermaid ` syntax. But I have commented out the  diagram for now. It's outdated, as you already mentioned. And I can't make much sense of it anyway.

I've tried to update the contents to the best of my knowledge and fixed the copy'n paste error  for the variable names.

**Review by micbar — APPROVED** [review]

**Review by mmattel — CHANGES_REQUESTED** [review]
While reviewing, the following catched my eyes, copying text snipplets for the first two items:

* `metadata`: The default. This backend persists the settings data via the `storage-system` service.
* ...the settings service caches the results of queries to the `system-storage` for providing faster responses.
* When checking the `storage-system` service, this service has caching implemented.

Reading the items one will ask: why is there caching in the `settings` service if data is handed over to the `storage-system` service that already has caching? We must provide a sufficient answer for this.

**Comment by mmattel** [comment]
Clarified with @rhafer what it is about, writing a small clarification, commit coming asap.

**Review by mmattel — APPROVED** [review]

**Comment by mmattel** [comment]
I would like to ask @EParzefall for a language review as this is needed in the admin docs anyways.

File: services/settings/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# Settings
+
+The `settings` service provides functionality for other services to register new settings as well as storing and retrieving the respective settings values.
```
Comment (EParzefall) [inline services/settings/README.md:3]:
```suggestion
The `settings` service provides functionality for other services to register new settings as well as storing and retrieving the respective settings' values.
```

File: services/settings/README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+The settings service is currently used for managing the:
+
+*   users' `profile` settings like the language and the Email notification settings,
```
Comment (EParzefall) [inline services/settings/README.md:7]:
```suggestion
*   users' `profile` settings like the language and the email notification settings,
```

File: services/settings/README.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+*   possible user roles and their respecitve permssions,
```
Comment (EParzefall) [inline services/settings/README.md:8]:
```suggestion
*   possible user roles and their respective permissions,
```

File: services/settings/README.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+*   assignment of roles to users.
+
+As an example, user profile settings that can be changed in the Web UI must be persisted.
```
Comment (EParzefall) [inline services/settings/README.md:11]:
```suggestion
As an example, user profile settings that can be changed in the Web UI must be persistent.  
```

File: services/settings/README.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+The settings service is currently used for managing the:
+
+*   users' `profile` settings like the language and the Email notification settings,
+*   possible user roles and their respecitve permssions,
+*   assignment of roles to users.
+
+As an example, user profile settings that can be changed in the Web UI must be persisted.  
+
+The settings service supports two different backends for persisting the data. The backend can be set using `SETTINGS_STORE_TYPE` environment variable. Supported values are:
```
Comment (EParzefall) [inline services/settings/README.md:13]:
```suggestion
The settings service supports two different backends for persisting the data. The backend can be set via the `SETTINGS_STORE_TYPE` environment variable. Supported values are:
```

File: services/settings/README.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    sdk --- sdks{ocis-settings<br>available?}
+    sdks ---|"yes"| os
+    sdks ---|"no"| defaults[Use set of<br>default values]
+    oa[oCIS services<br>e.g. ocis-accounts] ---|"saveSettingsBundle(bundle)"| os
+```
+-->
+
+## Caching
+
+When using `SETTINGS_STORE_TYPE=metadata`, the settings service caches the results of queries against the storage backend for providing faster responses. The content of this cache is independent of the cache used in the `storage-system` service as it caches directory listing and settings content stored in files.
```
Comment (EParzefall) [inline services/settings/README.md:39]:
```suggestion
When using `SETTINGS_STORE_TYPE=metadata`, the `settings` service caches the results of queries against the storage backend to provide faster responses. The content of this cache is independent of the cache used in the `storage-system` service as it caches directory listing and settings content stored in files.
```

File: services/settings/README.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+3.  The `settings` service can be scaled if not using `in-memory` stores and
+    the stores are configured identically over all instances.
+4.  When using `redis-sentinel`, the Redis master to use is configured via
+    `SETTINGS_CACHE_STORE_NODES` in the form of
+    `<sentinel-host>:<sentinel-port>/<redis-master>` like
+    `10.10.0.200:26379/mymaster`.
+
+## Settings Management
+
+oCIS services can register *settings bundles* with the settings service.
```
Comment (EParzefall) [inline services/settings/README.md:65]:
```suggestion
Infinite Scale services can register *settings bundles* with the settings service.
```

**Review by EParzefall — APPROVED** [review]

**Review by phil-davis — APPROVED** [review]

**Pull request merged**